lib/urbanairship/http_client.rb

```diff
@@ -0,0 +1,89 @@
+module Urbanairship
+  class HttpClient
+
+    # Make an HTTP Request
+    def self.request(*args)
+      HttpClient::Request.new(*args).send_request
+    end
+
+    class Response
+      attr_reader :body, :code, :headers
+
+      def initialize(http_response)
+        @body    = parse_json(http_response.body)
+        @code    = http_response.code.to_i
+        @headers = http_response.each_header.to_a
+      end
+
+      def [](attribute)
+        send(attribute) if respond_to?(attribute)
+      end
+
+      private
+
+      def parse_json(body)
+        JSON.parse(body.to_s)
+      rescue JSON::ParserError
+        {}
+      end
+    end
+
+    class Request
+      attr_reader :method, :uri, :headers, :parameters, :auth
+
+      def initialize(method, uri:, headers: {}, parameters: nil, auth: nil)
+        @method     = method.downcase.to_sym
+        @uri        = URI(uri)
+        @headers    = headers
+        @parameters = parameters
+        @auth       = auth
+        @request    = build_request
+      end
+
+      def send_request
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = true
+        http.read_timeout = 5
+
+        headers.each do |header, value|
+          @request[header] = value
+        end
+
+        if auth
+          @request.basic_auth(auth[:username], auth[:password])
+        end
+
+        begin
+          response = http.request(@request)
+        rescue Net::HTTPRequestTimeOut
+          raise 'Request timeout'
+        end
```

Comment:
@paneq Here is where we can handle exceptions—not sure which ones are important to handle.

Alternatively, I suppose it could use `RestClient` directly and capture the same exceptions Unirest does. IMO libraries should strive for as few dependencies as possible.